addon/components/o-s-s/infinite-select.ts

```diff
@@ -4,7 +4,16 @@ import { assert } from '@ember/debug';
 import { action } from '@ember/object';
 
 import { guidFor } from '@ember/object/internals';
-
+import type { SkinType } from './button';
+import type { IconNames } from './iconName.enum';
+
+export type InfiniteSelectAction = {
+  skin?: SkinType;
+  label?: string;
+  icon?: keyof typeof IconNames | string;
```

Comment:
just string imo. `IconNames` (trying to type all FontAwesome icons) was full of good intentions but it's not really do-able honestly 😅